moped-editor/src/components/GridTable/GridTable.js

```diff
@@ -439,95 +443,107 @@ const GridTable = ({
-                                      query.config.columns[
-                                        column
-                                      ].hasOwnProperty("className")
-                                        ? query.config.columns[column].className
-                                        : classes.tableCell
-                                    }
-                                  >
-                                    {query.isPK(column) ? (
-                                      // If there is custom JSX for the PK single item button, render it
-                                      (query.config.customSingleItemButton &&
-                                        query.config.customSingleItemButton(
-                                          row[column]
-                                        )) || (
-                                        <RouterLink
-                                          to={`/${query.singleItem}/${row[column]}`}
-                                        >
-                                          {query.config.columns[
-                                            column
-                                          ].hasOwnProperty("icon") ? (
-                                            <Icon
-                                              color={
-                                                query.config.columns[column]
-                                                  .icon.color
-                                              }
-                                            >
-                                              {
-                                                query.config.columns[column]
-                                                  .icon.name
-                                              }
-                                            </Icon>
-                                          ) : (
-                                            row[column]
-                                          )}
-                                        </RouterLink>
-                                      )
-                                    ) : query.config.columns[column]?.link ? (
-                                      query.getFormattedValue(
-                                        column,
-                                        buildLinkData(row, column)
-                                      )
-                                    ) : isAlphanumeric(column) ? (
-                                      <>
-                                        {query.config.columns[
+                      {data[query.table].length < 1 ? (
+                        <TableRow>
+                          <Box className={classes.noResults}>
+                            <Typography align="center">
+                              {query.config.noResultsMessage ??
+                                "No results found"}
+                            </Typography>
+                          </Box>
+                        </TableRow>
```

Comment:
This checks if there are no results and returns the appropriate message. 

The rest is indent changes because of the conditional.